util/config_stripper.py

```diff
@@ -78,11 +95,61 @@ def strip_tar(input, output):
     shutil.rmtree(tempdir)
     return 0
 
-
-def strip_config(path):
+def strip_layer(path):
+    # The original layer tar is of the form <random string>/layer.tar, the
+    # working directory is one level up from where layer.tar is.
+    original_dir = os.path.normpath(os.path.join(os.path.dirname(path), '..'))
+
+    buf = cStringIO.StringIO()
+
+    # Go through each file/dir in the layer
+    # Set its mtime to 0
+    # If it's a file, add its content to the running buffer
+    # Add it to the new gzip'd tar.
+    with tarfile.open(name=path, mode='r') as it:
+      with tarfile.open(fileobj=buf, mode='w') as ot:
+        ti = it.next()
+        while ti is not None:
+          ti.mtime = 0
+          if ti.isfile():
+            f = it.extractfile(ti)
+            ot.addfile(ti, f)
+          else:
+            ot.addfile(ti)
+          ti = it.next()
+
+    # Create the new diff_id for the config
+    tar = buf.getvalue()
+    diffid = hashlib.sha256(tar).hexdigest()
+    diffid = 'sha256:%s' % diffid
+
+    # Compress buf to gz
+    gzip_process = subprocess.Popen(
```

Comment:
I assume you're doing it this way for performance? Can you add a comment explaining why we're not using python's gzip?